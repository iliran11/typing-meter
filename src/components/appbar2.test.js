import React from 'react';
import AppBar from './appbar2';
import { shallow } from 'enzyme';

function AppBarFactory(walkThroughIconStatus) {
  return shallow(<AppBar walkThroughIconStatus={walkThroughIconStatus} />);
}
function getInfoIconStyle(walkThroughIconStatus) {
  const wrapper = AppBarFactory(walkThroughIconStatus);
  return wrapper.find('.walkthrough').prop('style');
}
describe('`AppBar`', () => {
  it('renders properly', () => {
    const wrapper = AppBarFactory();
    expect(wrapper).toMatchSnapshot();
  });
  describe(`help icon`, () => {
    it('is greyed out when icon status is false', () => {
      const informationIconStyle = getInfoIconStyle(false);
      expect(informationIconStyle).toHaveProperty('opacity', 0.5);
    });
    it('is not greyed when icon status is true', () => {
      const informationIconStyle = getInfoIconStyle(true);
      expect(informationIconStyle).toEqual({});
    });
  });
});
