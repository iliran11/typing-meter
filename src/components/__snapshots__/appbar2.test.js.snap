// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`AppBar\` renders properly 1`] = `
<div
  className="app-bar"
>
  <div
    className="left-hand"
  >
    <Link
      replace={false}
      to="/"
    >
      <Logo />
    </Link>
  </div>
  <div
    className="right-hand"
  >
    <Link
      replace={false}
      to="/settings"
    >
      <i
        className="fas fa-cog"
      />
    </Link>
    <i
      className="fab fab fa-github github"
    />
    <i
      className="fas fa-info-circle"
      onClick={null}
      style={
        Object {
          "opacity": 0.5,
        }
      }
    />
  </div>
</div>
`;
